sdk/core/src/policies/telemetry_policy.rs

```diff
@@ -47,6 +53,23 @@ impl Default for TelemetryPolicy {
     }
 }
 
+trait Environment {
```

Comment:
I'm not sure I get the utility of having this trait. At first I thought maybe this would be used to avoid having the tests tied to the environment variables, but that's not the case. I think it would just be easier to not have a trait for this.

Reply:
That's exactly what the trait is doing. See down in the test mod for mocking that the env vars are missing, as you called out in a previous comment. Can't reliably set and unset env vars since the tests can run in parallel according to docs. This is effectively what I've done in many other projects.

Reply:
Well yes, but you're _also_ testing against the actual env vars in another test, so you're not really avoiding the issue, you're just testing that this mocking capability that is only used in the tests. This doesn't seem helpful.

Reply:
I'm mocking to test that "unknown" is returned if the env vars aren't set and asserting the expected output. I don't see how that's any different from unit tests which, for example, test simple math operations to make sure no regressions occur down the road.

Yes, some of the tests are expecting the env vars - hence my comment about assuming cargo + rustc - but if you prefer I can have those return explicit values. That seemed unnecessary, given they are tests and we can probably safely assume cargo + rustc are being used. I appreciate your point about the production code maybe not using one or both.

Reply:
This has been completely refactored. See my comment below for details.